src/Parser.js

```diff
@@ -5,7 +5,11 @@ const glob = require('glob');
 
 class Parser {
   constructor(options = {}) {
-    this.extensions = {};
+    /**
+     * A map of extension names to either path to Asset class or Asset class itself
+     * @type {Map<string, string|Asset}
+     */
+    this.extensions = new Map();
```

Comment:
@DeMoorJasper this is the core of the change. I'll clean up unnecessary changes momentary

Reply:
I see you changed it into a map but wondering what issue this fixes?